engine/menu/field_debug_menu.asm

```diff
@@ -0,0 +1,518 @@
+	dw FieldDebug_InitToolGear
+	dw FieldDebug_ClearSetEventFlags
+	dw FieldDebug_RecoverPokemon
+	dw FieldDebug_InitCableClub
+	dw FieldDebug_NextFieldDebugMenuPage
+	dw FieldDebug_NPCMovementTest
+	dw FieldDebug_TogglePokemonFollowing
+	dw FieldDebug_PlayerFollowNPCTest
+	dw FieldDebug_Warp
+	dw FieldDebug_FieldCutTest
+	dw FieldDebug_CheckTile
+	dw FieldDebug_MoveToEntrance
+	dw FieldDebug_ToggleNPCMovement
+	dw FieldDebug_InitMapViewer
+	dw FieldDebug_ItemTest
+	dw FieldDebug_AccessPC
+	dw FieldDebug_AccessPokeMart
+	dw FieldDebug_SetTeleportLocation
+	dw FieldDebug_InitVRAMViewer
+	dw FieldDebug_OpenPokeGear
+
+RESET_GAME       EQU 0  ; unused
+CLOSE_MENU       EQU 1
+CHANGE_BORDER    EQU 2  ; unused
+GAME_SELECT      EQU 3
+CHANGE_TRANSPORT EQU 4
+CHANGE_TILESET   EQU 5  ; unused
+SHOW_MAP         EQU 6
+CHAR_VIEWER      EQU 7
+NAME_ENTRY       EQU 8  ; unused
+TOOL_GEAR        EQU 9
+EVENT_FLAGS      EQU 10
+RECOVER_PKMN     EQU 11
+CABLE_CLUB       EQU 12 ; unused
+NEXT_PAGE        EQU 13
+NPC_MOVE_TEST    EQU 14 ; unused - usable in bottom-left house in Silent Hills
+PKMN_FOLLOW      EQU 15 ; unused
+FOLLOW_NPC       EQU 16
+WARP             EQU 17
+CUT_TEST         EQU 18 ; unused
+CHECK_TILE       EQU 19 ; unused
+MOVE_ENTRANCE    EQU 20 ; unused - usable in Route 1
+NPC_MOVE_TOGGLE  EQU 21 ; unused
+MAP_VIEWER       EQU 22
+ITEM_TEST        EQU 23 ; unused
+ACCESS_PC        EQU 24
+ACCESS_MART      EQU 25
+SET_TELEPORT     EQU 26
+VRAM_VIEWER      EQU 27
+POKE_GEAR        EQU 28
+
+FieldDebugMenuPages:
+
+FieldDebugMenuPage1: ; 3F:40CE
+	db 7
+	db NEXT_PAGE
+	db WARP
+	db CHAR_VIEWER
+	db CHANGE_TRANSPORT
+	db TOOL_GEAR
+	db ACCESS_PC
+	db CLOSE
+	db -1
+
+FieldDebugMenuPage2: ; 3F:40D7
+	db 7
+	db NEXT_PAGE
+	db ACCESS_MART
+	db RECOVER_PKMN
+	db POKE_GEAR
+	db GAME_SELECT
+	db MAP_VIEWER
+	db CLOSE
+	db -1
+
+FieldDebugMenuPage3: ; 3F:40E0
+	db 7
+	db NEXT_PAGE
+	db EVENT_FLAGS
+	db VRAM_VIEWER
+	db SET_TELEPORT
+	db FOLLOW_NPC
+	db SHOW_MAP
+	db CLOSE
+	db -1
+
+SFX_MENU EQU 3 ; Menu opening sound
+
+FieldDebugMenu:: ; 3F:40E9
+	call RefreshScreen
+	ld de, SFX_MENU
+	call PlaySFX
+	ld hl, FieldDebugMenuHeader
+	call LoadMenuHeader
+	ld a, $00
+	ld [wFieldDebugMenuPage], a
+
+.Reopen: ; 3F:40FD
+	call UpdateTimePals
+	call UpdateSprites
+	ld a, [wFieldDebugMenuCursorBuffer]
+	ld [wMenuCursorBuffer], a
+	call OpenMenu
+	jr c, .Exit
+	ld a, [wMenuCursorBuffer]
+	ld [wFieldDebugMenuCursorBuffer], a
+	call PlaceHollowCursor
+	ld a, [wMenuJoypad]
+	cp A_BUTTON
+	jr z, .RunFunction
+	call ChangeFieldDebugMenuPage
+	jr .RunReturn
+
+.RunFunction: ; 3F:4123
+	ld a, [wMenuSelection]
+	ld hl, FieldDebugJumpTable
+	call CallJumptable
+
+; Field Debug menu items may have different return functions.
+.RunReturn: ; 3F:412C
+	ld hl, FieldDebugMenuReturns
+	jp CallJumptable
+
+.FieldDebugMenuReturns: ; 3F:4132
+	dw .Reopen
+	dw .Exit
+	dw .Close
+	dw .Clean
+	dw .Return
+
+.Exit: ; 3F:413C
+	call GetJoypad
+	ldh a, [hJoyDown]
+	bit 0, a
```

Comment:
Use the correct button constant. I don't remember which is bit 0 nor what the constants are called by heart, but something like `bit BUTTON_A, a`. Look around in the constants files, if this constant doesn't exist yet, make it.